### PR TITLE
Add deny_permissions, deny_platforms and display_name setting support

### PR DESCRIPTION
**Pull request by RicardoMusch**
Adds support for setting:

- display_name
As great as 'Set Modules' sounds... someone might want to change it up...

- deny_permissions
Not everyone likes to give users the ability to change environement setups ;)

- deny_platforms
Handy...

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -55,6 +55,43 @@ class CpenvApplication(sgtk.platform.Application):
     ModuleSpecSet = ModuleSpecSet
 
     def init_app(self):
+
+        try:
+            # Check if we should register this app by comparing
+            # deny_permissions to the user's permission group
+            # deny_permissions via info.yml only works for tk-shotgun
```
Comment (danbradham) [inline app.py]: If the intention is to only deny permissions to the Set Modules dialog, then I think this conditional block needs to move down to just before the UI dialog command is registered. That would ensure the app is still initialized and prepared to run when launching applications.

**Review by danbradham — COMMENTED** [review]
Thanks Ricardo, these are great additions.

File: app.py (hunk 1 above)
Comment (RicardoMusch) [reply app.py]: Ah I see, yes its intention is to only deny the app.
Will update!

Comment (RicardoMusch) [reply app.py]: Moved it down!

**Comment by danbradham** [comment]
Just noticed one last thing. The before_app_launch method will still execute on all platforms. It would be great if there was a permissions check at the beginning of before_app_launch to return early if the user is on a denied platform.

**Comment by RicardoMusch** [comment]
> Just noticed one last thing. The before_app_launch method will still execute on all platforms. It would be great if there was a permissions check at the beginning of before_app_launch to return early if the user is on a denied platform.

However we do want that right?
We just want to disallow the Set Module app from starting.

Or should deny_platforms encompass the overall setting that CPENV as a whole is denied on that platform?

**Comment by danbradham** [comment]
You know what, I think it would make sense to rename these to deny_ui_permissions and deny_ui_platforms. I think restricting cpenv to certain platforms would make sense potentially, but that could be handled in other ways. Like adding a platforms field to the Environment entity and exposing that in the Set Modules dialog. So let's keep this PR as you originally intended.

**Comment by RicardoMusch** [comment]
> You know what, I think it would make sense to rename these to deny_ui_permissions and deny_ui_platforms. I think restricting cpenv to certain platforms would make sense potentially, but that could be handled in other ways. Like adding a platforms field to the Environment entity and exposing that in the Set Modules dialog. So let's keep this PR as you originally intended.

Done!

**Comment by danbradham** [comment]
Excellent. Thanks for contributing!

**Comment by RicardoMusch** [comment]
> Excellent. Thanks for contributing!

Thanks for being my first ever pull request :P

**Pull request merged**